### PR TITLE
feat(components): Add a KServe component

### PR DESCRIPTION
**Pull request by markwinter**
**Description of your changes:**
I have added a KServe component for KServe 0.7.0

I have added it in a separate directory so the existing KFServing component can be accessed easily still because it is still used in the current Kubeflow releases.

Two questions:
- This component was made by  copying the KFServing component and updating it for KServe. I didn't copy the `OWNERS` file because I don't know what the etiquette is here. Should I just keep the same `OWNERS` file?
- Currently the `component.yaml` still points to the old image `quay.io/aipipeline/kfserving-component:v0.5.1`. This should be updated somehow but this should be done by a Kubeflow member with access to the right docker repo right?

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by Tomcli** [comment]
> * This component was made by  copying the KFServing component and updating it for KServe. I didn't copy the `OWNERS` file because I don't know what the etiquette is here. Should I just keep the same `OWNERS` file?

Yes, we need the `OWNERS` file in order to approve and assign us to review PRs for KServe component

**Comment by Tomcli** [comment]
> * Currently the `component.yaml` still points to the old image `quay.io/aipipeline/kfserving-component:v0.5.1`. This should be updated somehow but this should be done by a Kubeflow member with access to the right docker repo right?

Right now we are updating these image manually because community members don't have permission to add webhook to auto-update these images in their infra.

**Comment by markwinter** [comment]
@Tomcli 
Thanks for answering.

I've added the `OWNERS` file from the original KFServing component

**Comment by Tomcli** [comment]
@jagadeeshi2i are you still working on https://github.com/kubeflow/pipelines/pull/6716? 
We want to make sure your work get credited as well.

**Comment by marrrcin** [comment]
I've cloned the repo from @markwinter 's PR and build the image manually - I can confirm that the deployment works using combination of KServe 0.7 (with Knative/Istio) + Kubeflow 1.6.0, provided that the service account running the Kubeflow Pipeline has required permission to create resources in the `serving.kserve.io` API group.

**Comment by Tomcli** [comment]
/lgtm
I also updated the kserve image. In the future we can try to see can we add github actions to pull external PR/repo, so this process can be semi-automated.

File: components/kserve/component.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  - {name: Watch Timeout,             type: String, default: '300',        description: "Timeout seconds for watching until InferenceService becomes ready."}
+  - {name: Min Replicas,              type: String, default: '-1',         description: 'Minimum number of InferenceService replicas'}
+  - {name: Max Replicas,              type: String, default: '-1',         description: 'Maximum number of InferenceService replicas'}
+  - {name: Request Timeout,           type: String, default: '60',         description: "Specifies the number of seconds to wait before timing out a request to the component."}
+
+outputs:
+  - {name: InferenceService Status,   type: String,                        description: 'Status JSON output of InferenceService'}
+implementation:
+  container:
+    image: quay.io/aipipeline/kfserving-component:v0.5.1  # TODO: Update this
```
Comment (Tomcli) [inline components/kserve/component.yaml:24]: I updated this image registry
```suggestion
    image: quay.io/aipipeline/kserve-component:v0.7.0
```

**Comment by markwinter** [comment]
@Tomcli Great, thank you

**Comment by Tomcli** [comment]
@animeshsingh can you approve this PR? thanks

**Comment by markwinter** [comment]
Hi @sinachavoshi

Could you take a look at this PR when you have time please? Thanks!

**Comment by IronPan** [comment]
Thanks @markwinter for the contribution. 
Regarding the location, this component can resides at the top level directory if it includes a README.md that states the following

- Organization name and description
- Version information, following the [major].[minor].[patch] format.
- Test status. Assert the component is continuously tested by a CI infrastructure that the owner chooses.
- Owners information, including the organization or personal contact emails

Alternatively, it can reside in https://github.com/kubeflow/pipelines/tree/master/components/contrib

**Comment by markwinter** [comment]
@IronPan 

Thanks for taking a look.
I've updated the README to include the information you asked for.

**Comment by SinaChavoshi** [comment]
I'm no longer actively working on this project, assigning to IronPan.

**Comment by IronPan** [comment]
LGTM this since it states the test status as "manually tested". Recommend update the test status asap when it's in place.

**Comment by IronPan** [comment]
Also if it's challenging to test this if this code resides in KFP repo, a way to test this is to have a periodical(for example daily) testing, load_component_from_url() the latest component YAML from the HEAD, and test in your own hosted CI environment. That way the owner will aware of the bad PRs asap.

**Comment by markwinter** [comment]
@IronPan 
Thank for reviewing. I'll see if we can get the proper testing in place quickly.

I also created another PR for that appears to be causing the test failure
https://github.com/kubeflow/pipelines/pull/7042

**Pull request merged**